include/directx/d3dx12.h

```diff
@@ -4037,6 +4038,421 @@ class CD3DX12_NODE_MASK_SUBOBJECT
+
+        HRESULT result = m_pDevice->CheckFeatureSupport(D3D12_FEATURE_FORMAT_INFO, &dFormatInfo, sizeof(D3D12_FEATURE_DATA_FORMAT_INFO));
+
+        PlaneCount = dFormatInfo.PlaneCount;
+        return result;
+    }
+
+    // 6: GPU Virtual Address Support
+    // TODO: Check if it's exactly the same field as the one in the D3D12_Options feature
+    // FEATURE_SUPPORT_GET(UINT, m_dGPUVASupport, MaxGPUVirtualAddressBitsPerResource);
+    FEATURE_SUPPORT_GET(UINT, m_dGPUVASupport, MaxGPUVirtualAddressBitsPerProcess);
+
+    // 7: Shader Model
+    D3D_SHADER_MODEL HighestShaderModel()
+    {
+        return m_dShaderModel.HighestShaderModel;
+    }
+
+    // 8: D3D12 Options1
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions1, WaveOps);
+    FEATURE_SUPPORT_GET(UINT, m_dOptions1, WaveLaneCountMin);
+    FEATURE_SUPPORT_GET(UINT, m_dOptions1, WaveLaneCountMax);
+    FEATURE_SUPPORT_GET(UINT, m_dOptions1, TotalLaneCount);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions1, ExpandedComputeResourceStates);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions1, Int64ShaderOps);
+
+    // 10: Protected Resource Session Support
+    D3D12_PROTECTED_RESOURCE_SESSION_SUPPORT_FLAGS ProtectedResourceSessionSupport(UINT NodeIndex = 0) const
+    {
+        return m_dProtectedResourceSessionSupport[NodeIndex].Support;
+    }
+
+    // 12: Root Signature
+    D3D_ROOT_SIGNATURE_VERSION HighestRootSignatureVersion() const
+    {
+        return m_dRootSignature.HighestVersion;
+    }
+
+    // 16: Architecture1
+    // Same data fields can be queried from m_dArchitecture
+    FEATURE_SUPPORT_GET_NODE_INDEXED(BOOL, m_dArchitecture1, TileBasedRenderer);
+    FEATURE_SUPPORT_GET_NODE_INDEXED(BOOL, m_dArchitecture1, UMA);
+    FEATURE_SUPPORT_GET_NODE_INDEXED(BOOL, m_dArchitecture1, CacheCoherentUMA);
+    FEATURE_SUPPORT_GET_NODE_INDEXED(BOOL, m_dArchitecture1, IsolatedMMU);
+
+    // 18: D3D12 Options2
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions2, DepthBoundsTestSupported);
+    FEATURE_SUPPORT_GET(D3D12_PROGRAMMABLE_SAMPLE_POSITIONS_TIER, m_dOptions2, ProgrammableSamplePositionsTier);
+
+    // 19: Shader Cache
+    FEATURE_SUPPORT_GET_NAME(D3D12_SHADER_CACHE_SUPPORT_FLAGS, m_dShaderCache, SupportFlags, ShaderCacheSupportFlags);
+
+private: // Private helpers
+    
+    // Helper function to decide the highest shader model supported by the system
+    // Stores the result in m_dShaderModel
+    // Must be updated whenever a new shader model is added to the d3d12.h header
+    HRESULT QueryHighestShaderModel()
+    {
+        // Check support in descending order
+        HRESULT result;
+
+        // SHADER_MODEL_6_X
+        for (int i = D3D_SHADER_MODEL_6_7; i >= D3D_SHADER_MODEL_6_0; i--) {
+            m_dShaderModel.HighestShaderModel = static_cast<D3D_SHADER_MODEL>(i);
+            result = m_pDevice->CheckFeatureSupport(D3D12_FEATURE_SHADER_MODEL, &m_dShaderModel, sizeof(D3D12_FEATURE_DATA_SHADER_MODEL));
+            if (SUCCEEDED(result) || result != E_INVALIDARG) {

```

Comment:
maybe simplify this to if (result != E_INVALIDARG)?

Reply:
Resolved